wren-ui/src/apollo/server/backgrounds/dashboardCacheBackgroundTracker.ts

```diff
@@ -0,0 +1,194 @@
+  private projectService: IProjectService;
+  private deployService: IDeployService;
+  private queryService: IQueryService;
+  private runningJobs = new Set<number>();
+
+  constructor({
+    dashboardRepository,
+    dashboardItemRepository,
+    dashboardCacheRefreshRepository,
+    projectService,
+    deployService,
+    queryService,
+  }: {
+    dashboardRepository: IDashboardRepository;
+    dashboardItemRepository: IDashboardItemRepository;
+    dashboardCacheRefreshRepository: IDashboardCacheRefreshRepository;
+    projectService: IProjectService;
+    deployService: IDeployService;
+    queryService: IQueryService;
+  }) {
+    this.dashboardRepository = dashboardRepository;
+    this.dashboardItemRepository = dashboardItemRepository;
+    this.dashboardCacheRefreshRepository = dashboardCacheRefreshRepository;
+    this.projectService = projectService;
+    this.deployService = deployService;
+    this.queryService = queryService;
+    this.intervalTime = 60000; // 1 minute
+    this.start();
+  }
+
+  private start(): void {
+    logger.info('Dashboard cache background tracker started');
+    setInterval(() => {
+      this.checkAndRefreshCaches();
+    }, this.intervalTime);
+  }
+
+  private async checkAndRefreshCaches(): Promise<void> {
+    try {
+      // Get all dashboards with cache enabled
+      const dashboards = await this.dashboardRepository.findAllBy({
+        cacheEnabled: true,
+      });
+
+      for (const dashboard of dashboards) {
+        if (!dashboard.scheduleCron || !dashboard.nextScheduledAt) {
+          continue;
+        }
+
+        const now = new Date();
+        const nextScheduledAt = new Date(dashboard.nextScheduledAt);
+
+        // Check if it's time to refresh
+        if (now >= nextScheduledAt) {
+          logger.info(`Start Refreshing cache for dashboard ${dashboard.id}`);
+          await this.refreshDashboardCache(dashboard);
+          logger.info(
+            `Finished Refreshing cache for dashboard ${dashboard.id}`,
+          );
+        }
+      }
+    } catch (error) {
+      logger.error(`Error checking dashboard caches: ${error.message}`);
+    }
+  }
+
+  private async refreshDashboardCache(dashboard: any): Promise<void> {
+    if (this.runningJobs.has(dashboard.id)) {
+      logger.debug(`Dashboard ${dashboard.id} refresh already in progress`);
+      return;
+    }
+
+    this.runningJobs.add(dashboard.id);
+
+    try {
+      // Get all items for this dashboard
+      const items = await this.dashboardItemRepository.findAllBy({
+        dashboardId: dashboard.id,
+      });
+
+      // Get project and deployment info
+      const project = await this.projectService.getCurrentProject();
+      const deployment = await this.deployService.getLastDeployment(project.id);
+      const mdl = deployment.manifest;
+      const hash = uuidv4();
+
+      // Refresh cache for each item
+      for (const item of items) {
```

Comment:
I'm thinking that should we do a batch refresh ? maybe use `Promise.all` but in batches. Could be an enhancement next.